decidim-proposals/spec/services/decidim/proposals/proposal_serializer_spec.rb

```diff
@@ -228,6 +241,101 @@ module Proposals
+          expect(serialized).to include(created_in_meeting: proposal.created_in_meeting)
+        end
+
+        it "serializes the cost of the proposal" do
+          expect(serialized).to include(cost: proposal.cost)
+        end
+
+        it "serializes the cost report of the proposal" do
+          expect(serialized).to include(cost_report: proposal.cost_report)
+        end
+
+        it "serializes the execution period of the proposal" do
+          expect(serialized).to include(execution_period: proposal.execution_period)
+        end
+
+        context "when the votes are hidden" do
+          let!(:component) { create(:proposal_component, :with_votes_hidden) }
+          let!(:proposal) { create(:proposal, component:) }
+
+          it "does not include total count of votes" do
+            expect(serialized).to include(votes: nil)
+          end
+        end
+
+        # This is an internal field for admins which should not be published
+        context "when proposal notes count are hidden" do
+          it "does not publish them" do
+            expect(serialized).not_to include(proposal_notes_count: proposal.proposal_notes_count)
+          end
+        end
+
+        # This is an internal field for admins which should not be published
+        context "when valuation assignments are hidden" do
+          it "does not publish them" do
+            expect(serialized).not_to include(valuation_assignments_count: proposal.valuation_assignments_count)
+          end
+        end
+
+        context "when proposals have execution periods that are not published" do
+          let!(:proposal) { create(:proposal) }
+          let(:execution_period) { proposal.execution_period }
+
+          before do
+            proposal.update!(execution_period:, state_published_at: nil)
+          end
+
+          it "includes the execution period on a proposal not published" do
+            expect(serialized).to include(
+              execution_period:,
+              state_published_at: nil
+            )
+          end
+        end
+
+        context "when proposals have cost reports that are not published" do
+          let!(:proposal) { create(:proposal) }
+          let(:cost_report) { proposal.cost_report }
+
+          before do
+            proposal.update!(cost_report:, state_published_at: nil)
+          end
+
+          it "includes the cost report with a proposal not published" do
+            expect(serialized).to include(
+              cost_report:,
+              state_published_at: nil
+            )
+          end
+        end
+
+        context "when proposals with costs that are not published" do
+          let!(:proposal) { create(:proposal) }
+          let(:cost) { proposal.cost }
+
+          before do
+            proposal.update!(cost:, state_published_at: nil)
+          end
+
+          it "includes costs with a proposal not published" do
+            expect(serialized).to include(
+              cost:,
```

Comment:
Let's unify all the cost related fields togehter in a single example. Also, I think they're actually being serialized, and they should be `nil` in all the cases 
```suggestion
              cost: nil,
              cost_report: nil,
              execution_period: nil,
              answer: nil,
```